socorro/processor/processor_app.py

```diff
@@ -349,49 +241,42 @@ def process_crash(self, crash_id, ruleset_name, tmpdir):
 
         self.logger.info("completed %s", crash_id)
 
-    def _setup_source_and_destination(self):
+    def _set_up_source_and_destination(self):
         """Instantiate classes necessary for processing."""
-        self.queue = self.config.queue.crashqueue_class(
-            self.config.queue,
-            namespace=self.app_instance_name,
-        )
-        self.source = self.config.source.crashstorage_class(
-            self.config.source,
-            namespace=self.app_name,
-        )
-        self.destination = self.config.destination.crashstorage_class(
-            self.config.destination,
-            namespace=self.app_name,
-        )
-
-        self.processor = self.config.processor.processor_class(
-            config=self.config.processor, host_id=self.app_instance_name
-        )
-
-        self.temporary_path = self.config.processor.temporary_path
+        self.queue = build_instance_from_settings(settings.QUEUE)
+        self.source = build_instance_from_settings(settings.CRASH_SOURCE)
+        destinations = []
+        for key in settings.CRASH_DESTINATIONS_ORDER:
+            dest_obj = build_instance_from_settings(settings.CRASH_DESTINATIONS[key])
+            setattr(dest_obj, "crash_destination_name", key)
+            destinations.append(dest_obj)
+        self.destinations = destinations
+
+        self.pipeline = build_instance_from_settings(settings.PROCESSOR["pipeline"])
+
+        self.temporary_path = settings.PROCESSOR["temporary_path"]
         os.makedirs(self.temporary_path, exist_ok=True)
 
-        if self.config.companion_process.companion_class:
-            self.companion_process = self.config.companion_process.companion_class(
-                self.config.companion_process
-            )
-        else:
-            self.companion_process = None
-
-    def _setup_task_manager(self):
-        """instantiate the threaded task manager to run the producer/consumer
-        queue that is the heart of the processor."""
+    def _set_up_task_manager(self):
+        """Create and set up task manager."""
         self.logger.info("installing signal handers")
         # Set up the signal handler for dealing with SIGTERM. The target should be this
         # app instance so the signal handler can reach in and set the quit flag to be
         # True. See the 'respond_to_SIGTERM' method for the more information
         respond_to_SIGTERM_with_logging = partial(respond_to_SIGTERM, target=self)
         signal.signal(signal.SIGTERM, respond_to_SIGTERM_with_logging)
 
-        self.task_manager = self.config.producer_consumer.producer_consumer_class(
-            self.config.producer_consumer,
-            job_source_iterator=self.source_iterator,
-            task_func=self.transform,
+        # Create task manager
+        manager_class = settings.PROCESSOR["task_manager"]["class"]
+        manager_settings = settings.PROCESSOR["task_manager"]["options"]
```

Comment:
I thought about adding the ability to pass additional arguments to `build_instance_from_settings`, but I'm on the fence about it. We could use it here and another place.